x-pack/plugins/cases/server/client/cases/utils.ts

```diff
@@ -81,13 +81,29 @@ const getCommentContent = (comment: CommentResponse): string => {
   return '';
 };
 
-const countAlerts = (comments: CaseResponse['comments']): number =>
-  comments?.reduce<number>((total, comment) => {
-    if (comment.type === CommentType.alert || comment.type === CommentType.generatedAlert) {
-      return total + (Array.isArray(comment.alertId) ? comment.alertId.length : 1);
-    }
-    return total;
-  }, 0) ?? 0;
+interface CountAlertsInfo {
+  total: number;
+  pushed: number;
+  totalAlerts: number;
+}
+
+const countAlerts = (comments: CaseResponse['comments']): CountAlertsInfo => {
+  const countingInfo = { total: 0, pushed: 0, totalAlerts: 0 };
+  return (
+    comments?.reduce<CountAlertsInfo>(({ total, pushed, totalAlerts }, comment) => {
+      if (comment.type === CommentType.alert || comment.type === CommentType.generatedAlert) {
+        return {
+          // We could use ++total but total + 1 is more readable.
+          total: total + 1,
+          // We could use ++pushed but pushed + 1 is more readable.
```

Comment:
nit: I agree, but I don't think you need to comment this. Maybe in a Github comment but not in the code twice.

Reply:
Thanks for adding comments @cnasikas! I agree with @stephmilovic here, we could probably leave it out

Reply:
Ok!